src/components/ModalWithConfirmation.js

```diff
@@ -0,0 +1,37 @@
+import Modal from "./Modal.js";
+
+export default class ModalWithConfirmation extends Modal {
+  constructor(modalSelector, { submitButtonSelector, formSelector }) {
+    super(modalSelector);
+    this._button = this._modalElement.querySelector(submitButtonSelector);
+    this._callback = null;
+    this._form = this._modalElement.querySelector(formSelector);
+    this._originalButtonText = this._button.textContent;
+  }
+
+
+  renderingSaving(isSaving) {
+   if (isSaving) {
+     this._button.textContent = "Deleting...";
+   }  else {
+     this._button.textContent = this._originalButtonText;
+   }
+  }
+
+  setCallback(callback) {
```

Comment:
Great job implementing confirmation popup and the functionality related to it! In general, it can support any confirmation.